line_profiler/_line_profiler.pyx

```diff
@@ -339,10 +339,23 @@ cdef class LineProfiler:
                 entries += list(cmap[entry].values())
             key = label(code)
 
-            entries = [(e["lineno"], e["nhits"], e["total_time"]) for e in entries]
+            temp_entries = [(e["lineno"], e["nhits"], e["total_time"]) for e in entries]
             # If there are multiple entries for a line, this will sort them by increasing # hits
-            entries.sort()
-
+            temp_entries.sort()
+            # If a line causes a branch, the stats for the True branch and False branch are separate.
+            # So we merge them here with 2 passes
+            branch_entries = {}
+            # Find the indicies of items with the same line number
+            for indx, entry in enumerate(temp_entries):
+                branch_entries[entry[0]] = branch_entries.setdefault(entry[0], []) + [indx]
+            entries = []
+            for entry in branch_entries:
+                if len(branch_entries[entry]) < 2:
+                    # The entry is unique
+                    entries.append(temp_entries[branch_entries[entry][0]])
+                    continue
+                # A bit of an abomination, but it basically merges nhits and total_time from the duplicate entries
+                entries.append(tuple((x[0] if not indx else sum(x)) for indx, x in enumerate(zip(*[temp_entries[i] for i in branch_entries[entry]]))))
```

Comment:
It looks like the code immediately below here is deliberately keeping only the last entry for `for` loops etc, I think this fix will cause a regression where `for` lines show count N + 1

Reply:
It also shouldn't be too confusing if the first line of the for loop will probably show the correct count with the `if` bug fixed

Reply:
That's possible, I didn't notice a discrepancy when testing but it's possible that I was just tired or something. Thanks for your reviews!

Reply:
I have done some more testing and I don't see any discrepancy in the line hits column:
```py
Timer unit: 1e-06 s

Total time: 5.0375e-05 s
File: test2.py
Function: silly_example at line 1

Line #      Hits         Time  Per Hit   % Time  Line Contents
==============================================================
     1                                           @profile
     2                                           def silly_example():
     3         8          4.7      0.6      9.3      for i in range(7):
     4         7          3.5      0.5      7.0          if i % 2 == 0:
     5         4         32.7      8.2     64.8              print("even")
     6         3          0.7      0.2      1.4          elif i == 1:
     7         1          3.4      3.4      6.8              print("one")
     8                                                   else:
     9         2          5.4      2.7     10.8              print("three or five")
```

Reply:
Line 3 is the discrepency I was talking about, looking at https://github.com/pyutils/line_profiler/issues/210#issuecomment-1517360692 this currently shows 7 hits, i.e. not counting whatever pre-processing hit implicitly prepares the loop. All the lines inside the for block on the other hand are correct.

Reply:
Actually, could you take a closer look at the per-hits of the `print` lines? Those look too fast for a `print` statement to me, do you have a special setup?

Reply:
> Line 3 is the discrepancy I was talking about, looking at [#210 (comment)](https://github.com/pyutils/line_profiler/issues/210#issuecomment-1517360692) this currently shows 7 hits, i.e. not counting whatever pre-processing hit implicitly prepares the loop. All the lines inside the for block on the other hand are correct.

The issue is actually with the current version of line_profiler. Since the for loop has done 1 iteration when i is 0, 2 iterations when i is 1, 3 iterations when i is 2, and so on, we know that the hit number should be the number of iterations, which is i + 1 --> 7 + 1 --> 8.
> Actually, could you take a closer look at the per-hits of the print lines? Those look too fast for a print statement to me, do you have a special setup?

I don't believe I have a special setup. What kind of time taken were you expecting? I'm on a laptop running Ubuntu 22.04.2, with Linux kernel 6.1.13 and with turbo boost disabled. My hardware is an intel i7-11800H CPU, 32GB of RAM, a nvidia 3060 laptop GPU, and a 1TB NVMe SSD, if that matters. with Here's the print function looped 1000 times to try and reduce noise:
```py
Timer unit: 1e-06 s

Total time: 0.00394186 s
File: test3.py
Function: main at line 2

Line #      Hits         Time  Per Hit   % Time  Line Contents
==============================================================
     2                                           @profile
     3                                           def main():
     4      1001        220.7      0.2      5.6      for _ in range(1000):
     5      1000       3721.2      3.7     94.4          print("0123456789")
```

Reply:
> The issue is actually with the current version of line_profiler. Since the for loop has done 1 iteration when i is 0, 2 iterations when i is 1, 3 iterations when i is 2, and so on, we know that the hit number should be the number of iterations, which is i + 1 --> 7 + 1 --> 8.

That's incorrect. `range(7)` only loops over 0 through 6, so your i + 1 is 7 items. For example, in your last block you can see that the print line correctly gets 1000 hits but the `for` line is getting 1001 hits. I think this is slightly confusing for users, but I also think it is far less confusing for users than the `if`/`elif` bug, so I think merging this was fine. However, this does merit a new issue to see if this is reasonable to fix or at least document the regression; I'll open one.